crates/bevy_color/src/hsla.rs

```diff
@@ -97,6 +97,35 @@ impl Hsla {
         let hue = index.wrapping_mul(FRAC_U32MAX_GOLDEN_RATIO) as f32 * RATIO_360;
         Self::hsl(hue, 1., 0.5)
     }
+
+    /// Converts the color into a [f32; 4] array in HSLA order.
+    ///
+    /// This is useful for passing the color to a shader.
+    pub fn to_f32_array(&self) -> [f32; 4] {
```

Comment:
What's the reason for implementing this as a custom function instead of `impl From<Hsla> for [f32; 4]`?

Just curious if I'm missing something here, the latter would allow us to also implement it without the alpha for `[f33; 3]` and then we would have a "symmetric" implementation.

Reply:
there was already a custom function for linear_rgba, decided to copy it over to the other colors, this does seem to be a better approach